crates/cuda/src/lib.rs

```diff
@@ -104,65 +109,93 @@ impl SP1CudaProver {
     /// Creates a new [SP1Prover] that runs inside a Docker container and returns a
     /// [SP1ProverClient] that can be used to communicate with the container.
     pub fn new() -> Result<Self, Box<dyn StdError>> {
-        let container_name = "sp1-gpu";
-        let image_name = std::env::var("SP1_GPU_IMAGE")
-            .unwrap_or_else(|_| "public.ecr.aws/succinct-labs/moongate:v4.1.0".to_string());
-
-        let cleaned_up = Arc::new(AtomicBool::new(false));
-        let cleanup_name = container_name;
-        let cleanup_flag = cleaned_up.clone();
-
-        // Check if Docker is available and the user has necessary permissions
-        if !Self::check_docker_availability()? {
-            return Err("Docker is not available or you don't have the necessary permissions. Please ensure Docker is installed and you are part of the docker group.".into());
-        }
-
-        // Pull the docker image if it's not present
-        if let Err(e) = Command::new("docker").args(["pull", &image_name]).output() {
-            return Err(format!("Failed to pull Docker image: {}. Please check your internet connection and Docker permissions.", e).into());
-        }
-
-        // Start the docker container
-        let rust_log_level = std::env::var("RUST_LOG").unwrap_or_else(|_| "none".to_string());
-        Command::new("docker")
-            .args([
-                "run",
-                "-e",
-                &format!("RUST_LOG={}", rust_log_level),
-                "-p",
-                "3000:3000",
-                "--rm",
-                "--gpus",
-                "all",
-                "--name",
-                container_name,
-                &image_name,
-            ])
-            // Redirect stdout and stderr to the parent process
-            .stdout(Stdio::inherit())
-            .stderr(Stdio::inherit())
-            .spawn()
-            .map_err(|e| format!("Failed to start Docker container: {}. Please check your Docker installation and permissions.", e))?;
-
-        // Kill the container on control-c
-        ctrlc::set_handler(move || {
-            tracing::debug!("received Ctrl+C, cleaning up...");
-            if !cleanup_flag.load(Ordering::SeqCst) {
-                cleanup_container(cleanup_name);
-                cleanup_flag.store(true, Ordering::SeqCst);
+        let moongate_endpoint = std::env::var("SP1_MOONGATE_ENDPOINT").ok();
```

Comment:
Instead of using an environment variable, you should have a separate `with_moongate_endpoint` new function that initializes with this environment variable.

I'm generally opposed to adding environment flags in the library unless they're absolutely necessary, as they're more brittle. (e.g. with `SKIP_SIMULATION` and others, we switched to declarative approaches).